src/compiler/utilities.ts

```diff
@@ -10657,3 +10657,17 @@ export function isSyntacticallyString(expr: Expression): boolean {
     }
     return false;
 }
+
+/** @internal */
+export function isSyntacticallyNumeric(expr: Expression): boolean {

```

Comment:
Nit: I think the implementation of this function is correct, but the name is not. _Unlike_ what we settled on with `isSyntacticallyString`, what matters here _is_ actually that the value is evaluatable to a constant (with no extra-file info), because the emit feature being protected is the ability for the next member value to be equal to this member value plus one. So for example, the expression `+variableOfUnknownValue` is “syntactically numeric,” but does not and should not pass this function, because we don’t know the constant value.

Reply:
Good point. I suppose the "syntactic" part is still important because we don't want to allow sth like this in isolatedModules:

```ts
// a.ts
export const a = 1;
// b.ts
import {a} from './a';
enum Foo {A = a, B}
```

How about `syntacticallyEvaluatesToConstantNumber` or `isSyntacticallyNumericConstant`?

Given that this is different from `isSyntacticallyString`, would you prefer the function to live in the checker?

Reply:
I’m fine with either of those names, and I think if it doesn’t need to be used outside the checker, it can go in the checker.